### ISSUE TITLE
LIFX brightness 0 issue

### ISSUE THREAD

**Issue by commento** [getsenic/senic-hub#199]
The LIFX component perceives brightness 0 as a valid brightness.
If the brightness value is decreased till 0 with nuimo and the light is then turned-on after this, the light stays off. 
The unexpected behaviour is also that sometimes the light turn on even if brightness 0 is applied.
To be analyzed if the issue is in nuimo app or lifx home assistant component but the second option is more probable because it does not takes place with lightify and dali (light nuimo app component) through home assistant.
```
2017-06-26 12:59:24,060 DEBUG nuimo_app.components.light Is On: False, Brightness: None
2017-06-26 12:59:24,964 DEBUG nuimo_app Ignoring gesture event: Gesture.BUTTON_RELEASE
2017-06-26 13:00:33,697 DEBUG nuimo_app Processing gesture event: Gesture.BUTTON_PRESS
2017-06-26 13:00:33,700 DEBUG nuimo_app.components Call service light:turn_on with data: {'entity_id': 'light.lifx_bulb_203d10'}
2017-06-26 13:00:33,701 DEBUG nuimo_app.hass Sending request id: 378
2017-06-26 13:00:33,702 DEBUG nuimo_app.hass {'domain': 'light',
 'id': 378,
 'service': 'turn_on',
 'service_data': {'entity_id': 'light.lifx_bulb_203d10'},
 'type': 'call_service'}
2017-06-26 13:00:33,731 DEBUG nuimo_app.hass Received result id: 378 success: True result:
2017-06-26 13:00:33,732 DEBUG nuimo_app.hass None
2017-06-26 13:00:33,733 DEBUG nuimo_app.components Calling service light:turn_on succeeded with result: {'id': 378, 'result': None, 'success': True, 'type': 'result'}
2017-06-26 13:00:33,757 DEBUG nuimo_app.components.light Updated light state: {'attributes': {'brightness': 0,
                'color_temp': 400.0,
                'friendly_name': 'LIFX Bulb 203d10',
                'max_mireds': 500,
                'min_mireds': 154,
                'rgb_color': [0, 0, 0],
                'supported_features': 51},
 'entity_id': 'light.lifx_bulb_203d10',
 'last_changed': '2017-06-26T13:00:33.738737+00:00',
 'last_updated': '2017-06-26T13:00:33.738737+00:00',
 'state': 'on'}
2017-06-26 13:00:33,758 DEBUG nuimo_app.components.light Is On: True, Brightness: 0
2017-06-26 13:00:34,614 DEBUG nuimo_app Ignoring gesture event: Gesture.BUTTON_RELEASE
```

**Comment by commento** [comment]
Solved in home assistant lifx_legacy component at line #225:
```
        if ATTR_BRIGHTNESS in kwargs:
            brightness = kwargs[ATTR_BRIGHTNESS] * (BYTE_MAX + 1)
        else:
            brightness = self._bri
+            if brightness == 0:
+               brightness = 1
```
When the turn_on command without an associated brightness value is received from nuimo app the stored brightness is checked and, if the value equal is to 0 then it is changed to 1 (could it be changed to max brightness value).
I will test it and propose a PR to home assistant.

**Comment by commento** [comment]
Now it works but I noticed that reaching the brightness 0 with nuimo is still perceived as light-on.

So a better fix could be provided on the nuimo_app light component on_rotation() method and removing the previous correction.

From line #39:
```
    def on_rotation(self, delta):
        if (not self.is_on or (self.brightness or 0) <= 0) and delta <= 0:
            self.nuimo.display_matrix(matrices.LIGHT_OFF)
            return
        if self.brightness is None:
            self.brightness = 1
        else:
            self.brightness = min(max(int(self.brightness + 255 * delta), 0), 255)
+        if self.brightness == 0:
+            self.on_button_press()
+        else:
+            self.send_brightness()
        self.nuimo.display_matrix(matrices.progress_bar(self.brightness / 255.0), fading=True, ignore_duplicates=True)
```

**Comment by commento** [comment]
fixed with PR #201
